src/js/components/CoursesList/CoursesList.js

```diff
@@ -1,26 +1,52 @@
-import React from 'react';
+import React, { PureComponent } from 'react';
+import { getCourses } from '../../services';
+import { parseTime, parseDate } from '../../helpers';
 import styles from './CoursesList.less';
 import ListItem from '../ListItem';
+import Loader from '../Loader';
 import Button from '../Button';
-import template from './template';
 
 const renderList = temp => temp.map(item => (
-  <ListItem title={item.title} timing={item.timing} text={item.text} key={item.id} />));
+  <ListItem
+    title={item.name}
+    timing={parseTime(item.length)}
+    text={item.description}
+    key={item.id}
+    date={parseDate(item.date)}
+  />
+));
+class CoursesList extends PureComponent {
+  state = {
+    courses: [],
+    isFetching: true
+  }
 
-const CoursesList = () => (
-  <div className={styles.container}>
-    <span className={styles.header}>
-    Courses
-    </span>
-    {renderList(template)}
-    <Button
-      variant="contained"
-      color="primary"
-      fullWidth
-    >
-    Load more
-    </Button>
-  </div>
-);
+  componentDidMount() {
+    getCourses()
+      .then(courses => this.setState({ courses }))
+      .then(() => this.setState({ isFetching: false }));
+  }
+
+  render() {
+    return (
+      <div className={styles.container}>
+        <span className={styles.header}>
+        Courses
+        </span>
+        {this.state.isFetching && <Loader />}
```

Comment:
destructuring is your friend. don't forget about that)
btw, isFetching return <Loader/> before return more obvious then && expression inside main return

Reply:
fixed)